### PR TITLE
Make headers in help hover look prettier

### PR DESCRIPTION
**Pull request by ManuelHentschel**
Modifies the header of help hovers to remove the first line.
Information about the package is moved to the first proper header instead.

### CODE REVIEW

**Comment by renkun-ken** [comment]
The title might not look better than before if it is long.

image

**Comment by renkun-ken** [comment]
What about using 

```r
pattern <- "<table.*?<td>(.*?)\\s+{(.*?)}<\\/td>.*?<\\/table>\\n*"
replacement <- "<b>\\1</b> <i>{\\2}</i><hr/>"
```

image

Both topic and package are preserved in the headline of the hover.

**Comment by ManuelHentschel** [comment]
> What about using
> 
> ```r
> pattern <- "<table.*?<td>(.*?)\\s+{(.*?)}<\\/td>.*?<\\/table>\\n*"
> replacement <- "<b>\\1</b> <i>{\\2}</i><hr/>"
> ```

That looks quite nice as well. What bothers me a bit here (and in the original version) is the vertical space taken up by the large heading plus the line "Encoding {base}", which I tried to reduce by moving the package info into the heading.

What do you think about these?
``` r
replacement <- "<b>\\1</b> <i>{\\2}</i><br/>\\3<hr/>"
pattern <- "<table.*?<td>(.*?)\\s*{(.*?)}<\\/td>.*?<\\/table>\\n*<h2>\\s*(.*?)\\s*<\\/h2>"
```
image

**Comment by renkun-ken** [comment]
> What do you think about these?

It looks good! Let's use this one.

File: R/workspace.R [hunk 1]
```diff
@@ -157,6 +157,10 @@ Workspace <- R6::R6Class("Workspace",
                     if (requireNamespace("rmarkdown", quietly = TRUE) &&
                         rmarkdown::pandoc_available()) {
                         html <- enc2utf8(repr::repr_html(hfile))
+                        # Make header look prettier:
+                        pattern <- "<table.*?<td>(.*?)\\s*{(.*?)}<\\/td>.*?<\\/table>\\n*<h2>\\s*(.*?)\\s*<\\/h2>"
+                        replacement <- "<b>\\1</b> <i>{\\2}</i><br/>\\3<hr/>"
+                        html <- gsub(pattern, replacement, html, perl=TRUE)
```
Comment (renkun-ken) [inline R/workspace.R:163]: Please "Put spaces around all infix operators." here.

**Review by renkun-ken — APPROVED** [review]
LGTM

**Comment by renkun-ken** [comment]
Just noticed that pandoc < 2.0 will produce different result with the modified html:

image

Let me do a quick fix for this.

**Comment by renkun-ken** [comment]
pandoc >= 2.0

```
**get** *{base}*  \nReturn the Value of a Named Object\n\n-----\n\n
```

pandoc < 2.0

```
**get** *{base}*\n\nReturn the Value of a Named Object\n\n------------------------------------------------------------------------\n\n
```

Looks llike `<br/>` is not translated to `  \n`.

**Comment by renkun-ken** [comment]
Fixed via 62b06e5c7b21fc045de47cd62046eec08ffd2b67.

```r
replacement <- "<b>\\1</b> <i>{\\2}</i><p>\\3</p><hr/>"
```

image

**Pull request merged**